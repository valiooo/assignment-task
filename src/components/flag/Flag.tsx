import FlagEur from '../../svg/EUR.svg';
import FlagPln from '../../svg/PLN.svg';
import FlagGbp from '../../svg/GBP.svg';
import FlagUah from '../../svg/UAH.svg';

export type CurrencyProps = 'PLN' | 'EUR' | 'GBP' | 'UAH';

export const Flag = ({ currency }: { currency: CurrencyProps }) => {
  switch (currency) {
    case 'PLN':
      return <FlagPln data-testid="flag-PLN" />;
    case 'EUR':
      return <FlagEur data-testid="flag-EUR" />;
    case 'GBP':
      return <FlagGbp data-testid="flag-EUR" />;
    case 'UAH':
      return <FlagUah data-testid="flag-UAH" />;
    default:
      return null;
  }
};
