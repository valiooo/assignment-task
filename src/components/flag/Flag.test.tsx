import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

import { Flag } from './Flag';

describe('Flag', () => {
  test('should render Flag', () => {
    const { asFragment } = render(<Flag currency="EUR" />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('should render correct flag', () => {
    render(<Flag currency="PLN" />);

    expect(screen.getByTestId('flag-PLN')).toBeInTheDocument();
  });
});
