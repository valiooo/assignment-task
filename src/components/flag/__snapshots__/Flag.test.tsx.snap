// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flag should render Flag 1`] = `
<DocumentFragment>
  <svg
    data-testid="flag-EUR"
  />
</DocumentFragment>
`;
